joomla/joomla-cms
# [New Feature] Implement Redis Session Manager

This PR implements Redis session manage.

It works with redis server in TCP or Unix Socket.
### Test Instructions
- Install Redis server.
- Install Redis PHP pecl.
- Configure Redis server with tcp and/or unix socket.

redis.conf example for only TCP:

```
port 6379
```

redis.conf example for only Unix Socket:

```
port 0
unixsocket /var/run/redis/redis.sock
unixsocketperm 777
```

redis.conf example for TCP and Unix Socket:

```
port 6379
unixsocket /var/run/redis/redis.sock
unixsocketperm 777
```
- Restart redis and php services
- Check if redis is accepting connections

For TCP: `redis-cli ping`
For Unix Socket: `redis-cli -s /var/run/redis/redis.sock ping`
- Download joomla staging + redis session https://github.com/andrepereiradasilva/joomla-cms/archive/staging.zip
- Configure redis session manager in global configuration

For TCP:

```
Host: Redis server ipv4 or localhost or domain (ex: localhost)
Port: Redis server port (ex: 6379)
Persist: Yes or No
Auth: Empty or password
Database: integer
```

For Unix Socket:

```
Host: Redis socket path (ex: /var/run/redis/redis.sock)
Port: 0 or -1
Persist: Yes or No
Auth: Empty or password
Database: integer
```
- Try admin and frontend login and logout
### Tested in:
- CentOS Linux 7.1
  - redis v2.8.19-2 (epel repository)
  - php56 v5.6.9-1 (webtatic repository)
  - php56 pecl redis v2.2.7-1 (webtatic repository)
- Joomla 3.4.1
### More information
- https://github.com/phpredis/phpredis#php-session-handler
